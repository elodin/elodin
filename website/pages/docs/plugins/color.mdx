import DocLayout from '../../../components/DocLayout.js'

export default DocLayout

# Color

```bash
yarn add @elodin/plugin-color
```

The color plugin converts all used colors into a consistent color format.

## Usage

To use the plugin, simply add it to your Elodin configuration and you're ready to go.

```javascript name=elodin.config.js
import color from '@elodin/plugin-color'

module.exports = {
  plugins: [color()],
}
```

## Configuration

| Option | Type                           | Default | Description                      |
| ------ | ------------------------------ | ------- | -------------------------------- |
| format | `rgb`, `hsl`, `hex`, `keyword` | `rgb`   | The prefered output color format |

### Example

Let's say we want all colors to be written as `rgb` colors.

```elodin name=button.elo info=Input
style Button {
  backgroundColor: rgb(0 10 0)
  borderTopColor: white
}
```

```elodin name=button.elo info=Output
style Button {
  backgroundColor: rgb(0 10 0)
  borderTopColor: rgb(255 255 255)
}
```
