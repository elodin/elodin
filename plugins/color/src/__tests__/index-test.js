import { traverse, parse } from '@elodin/core'

import color from '../index'

describe('Normalizing colors', () => {
  it('should correctly set color formats', () => {
    const file = `style Button { backgroundColor: red borderBottomColor: rgb(255 100 50) }`

    const ast = parse(file).ast

    expect(traverse(ast, [color()])).toMatchSnapshot()
  })
})
