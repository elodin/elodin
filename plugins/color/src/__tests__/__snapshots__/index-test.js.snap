// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Normalizing colors should correctly set color formats 1`] = `
Object {
  "body": Array [
    Object {
      "body": Array [
        Object {
          "comments": Array [],
          "dynamic": false,
          "property": "backgroundColor",
          "raw": false,
          "type": "Declaration",
          "value": Object {
            "type": "Identifier",
            "value": "red",
          },
        },
        Object {
          "comments": Array [],
          "dynamic": false,
          "property": "borderBottomColor",
          "raw": false,
          "type": "Declaration",
          "value": Object {
            "callee": "rgb",
            "params": Array [
              Object {
                "negative": false,
                "type": "Integer",
                "value": 255,
              },
              Object {
                "negative": false,
                "type": "Integer",
                "value": 100,
              },
              Object {
                "negative": false,
                "type": "Integer",
                "value": 50,
              },
            ],
            "type": "FunctionExpression",
          },
        },
      ],
      "comments": Array [],
      "name": "Button",
      "type": "Style",
    },
  ],
  "type": "File",
}
`;
